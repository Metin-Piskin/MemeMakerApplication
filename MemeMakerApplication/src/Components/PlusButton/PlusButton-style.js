import { StyleSheet } from "react-native";

export default StyleSheet.create({
    container: {
        position: 'absolute',
        bottom: 15,
        right: 15,
        borderRadius: 100,
    },
    text: {
        paddingHorizontal: 10,
        paddingVertical: 10,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
    }
})