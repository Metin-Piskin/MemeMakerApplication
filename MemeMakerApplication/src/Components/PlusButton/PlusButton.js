import React from 'react';
import { TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import Feather from 'react-native-vector-icons/Feather';

import styles from './PlusButton-style';

const PlusButton = ({ onPress }) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            testID='plus-button-press'
        >
            <LinearGradient
                colors=
                {[
                    '#161718',
                    '#2C2C2E',
                ]}
                style={styles.container}
                testID='plus-button-container'
            >
                <Feather
                    name='plus'
                    size={40}
                    color='#C7C6CB'
                    style={styles.text}
                    testID='plus-button-text'
                />
            </LinearGradient>
        </TouchableOpacity>
    )
}

export default PlusButton;