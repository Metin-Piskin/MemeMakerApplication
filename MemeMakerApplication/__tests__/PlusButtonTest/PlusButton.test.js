import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import PlusButton from '../../src/Components/PlusButton';
import PlusButtonStyles from '../../src/Components/PlusButton/PlusButton-style';

jest.mock('react-native-linear-gradient', () => 'LinearGradient');
jest.mock('react-native-vector-icons/Feather', () => 'Feather');

test('renders correctly', () => {
    const comp = render(<PlusButton />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const mockFunction = jest.fn();
    const comp = render(<PlusButton onPress={mockFunction} />);

    const PlusButtonPress = comp.getByTestId('plus-button-press');
    fireEvent(PlusButtonPress, 'press');

    expect(mockFunction).toBeCalled();
});

test('should render default theme style', () => {
    const comp = render(<PlusButton />);
    const PlusButtonContainerStyle = comp.getByTestId('plus-button-container').props.style;

    expect(PlusButtonContainerStyle).toMatchObject(PlusButtonStyles.container);
});

test('should render default theme style', () => {
    const comp = render(<PlusButton />);
    const PlusButtonTextStyle = comp.getByTestId('plus-button-text').props.style;

    expect(PlusButtonTextStyle).toMatchObject(PlusButtonStyles.text);
});

