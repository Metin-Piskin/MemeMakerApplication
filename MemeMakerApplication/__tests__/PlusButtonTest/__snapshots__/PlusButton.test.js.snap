// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="plus-button-press"
>
  <LinearGradient
    colors={
      [
        "#161718",
        "#2C2C2E",
      ]
    }
    style={
      {
        "borderRadius": 100,
        "bottom": 15,
        "position": "absolute",
        "right": 15,
      }
    }
    testID="plus-button-container"
  >
    <Feather
      color="#C7C6CB"
      name="plus"
      size={40}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "justifyContent": "center",
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
      testID="plus-button-text"
    />
  </LinearGradient>
</View>
`;
